Added http client and server to the demo app to make the traces more interesting (#3701)
open-telemetry/opentelemetry-collector: examples/demo/README.md

*IMPORTANT:* This is a pre-released version of the OpenTelemetry Collector.

> This demo contains a simple client and server applications that use the

Reviewer: nit
```suggestion
This demo contains a client and server applications that use the
```

opentelemetry Go library for instrumentation and for sending telemetry data
to the opentelemetry collector.

> The client periodically makes a simple http call to the server, those calls
> create client and server spans and metrics track basic information like
> number of http requests and latencies.

Reviewer: ```suggestion
The client periodically makes http calls to the server which
create client spans, server spans and metrics that track information like
number of http requests and latency.
```
